File: distance_3d.m
function [leg] = distance_3d(pt_A, pt_B)

% calculate deltas
x_del = pt_B(1) - pt_A(1);
y_del = pt_B(2) - pt_A(2);
z_del = pt_B(3) - pt_A(3);

% calculate path length (the "leg" of the triangle)
leg = sqrt(x_del^2 + y_del^2 + z_del^2);

File: Motion_Generation_Python/distance_3d.py
def distance_3d(pt_A, pt_B):
    x_del = pt_B[0] - pt_A[0];
    y_del = pt_B[1] - pt_A[1];
    z_del = pt_B[2] - pt_A[2];

def leg():
    leg = (x_del**2 + y_del**2 + z_del**2)**(1/2);
    return(leg)
